core/src/com/mygdx/game/models/Command.java

```diff
@@ -1,14 +1,93 @@
 package com.mygdx.game.models;
 
-public interface Command {
+/**
+ * Class to represent a command
+ */
+public class Command {
+    /* Type of action the command represents */
+    private String actionType;
 
-    /**
-     * Get the action associated with the command
-     */
-    String getAction();
+    /* Acting unit's ID */
+    private int unitId;
 
-    /**
-     * Get position that the action is directed at
-     */
-    Position getPosition();
+    /* Target unit's ID */
+    private int targetUnitId;
+
+    /* Position action is taken from */
+    private Position startPosition;
+
+    /* Targeted position */
+    private Position targetPosition;
+
+    /* The number of turns the command has been active */
+    private int passedDuration;
+
+    /* How long the command takes */
+    private int totalDuration;
+
+    public Command(String actionType, int unitId, int targetUnitId, Position startPosition, Position targetPosition, int passedDuration, int totalDuration) {
+        this.actionType = actionType;
+        this.unitId = unitId;
+        this.targetUnitId = targetUnitId;
+        this.startPosition = startPosition;
+        this.targetPosition = targetPosition;
+        this.passedDuration = passedDuration;
+        this.totalDuration = totalDuration;
+    }
+
+    public String getActionType() {
+        return actionType;
+    }
+
+    public void setActionType(String actionType) {
+        this.actionType = actionType;
```

Comment:
We shouldn't need andy setters for this class. It should be immutable

Reply:
Fixed